engine/tests/ElggCoreRegressionBugsTest.php

```diff
@@ -224,6 +224,9 @@ public function test_parse_urls() {
 			'already anchor <a href="http://twitter.com/">twitter</a> test' =>
 				'already anchor <a href="http://twitter.com/">twitter</a> test',
 
+			'already anchor with link text <a href="http://twitter.com/">http://twitter.com</a> test' =>
```

Comment:
We need to make sure that regex is not greedy. Needs a test with more than one anchor tag. Maybe a mix of diffetent tags. And an anchor tag within another tag.